### ISSUE TITLE
check whether we can get disk I/O from the metrics API

### ISSUE THREAD

**Issue by umanwizard** [MaterializeInc/materialize#15995]
### Feature request

CC @jpepin , @hlburak

**Comment by benesch** [comment]
Why is this important? I don't think we should be exposing disk I/O stats to users—at least not until we have swap!

**Comment by umanwizard** [comment]
This doesn't say anything about exposing it to users, necessarily! Could be useful in Grafana.

FWIW, we decided in a meeting that this is low-pri.

**Comment by benesch** [comment]
Oh, why do we need it in the metrics server, then? Is this just a request for per-pod disk I/O in Grafana by whatever means necessary?

**Comment by umanwizard** [comment]
Yep, although having it in the metrics server would _also_ make it possible to expose to users, if that is indeed useful.

**Comment by teskje** [comment]
@umanwizard Is this still relevant? I remember that we discussed this a couple times and the result was that k8s doesn't expose disk metrics?

**Comment by umanwizard** [comment]
@guswynn can you decide if this is still useful for spill-to-disk work, and close otherwise?

**Comment by benesch** [comment]
The answer was "yes, we can get this from the metrics API via the [Prometheus adapter](https://github.com/kubernetes-sigs/prometheus-adapter)". 🎉  Implemented in https://github.com/MaterializeInc/cloud/pull/7009.
